client/web/src/search/stream.tsx

```diff
@@ -276,18 +272,22 @@ function toGQLCommitMatch(commit: CommitMatch): GQL.ICommitSearchResult {
     return gqlCommit as GQL.ICommitSearchResult
 }
 
+export type StreamingResultsState = 'loading' | 'error' | 'complete'
+
 export interface AggregateStreamingSearchResults {
+    state: StreamingResultsState
     results: GQL.SearchResult[]
     alert?: Alert
     filters: Filter[]
     progress: Progress
+    error?: Error
 }
```

Comment:
This change would make the exported type more type-safe to manipulate, by guaranteeing that the `error` property is always set for error results, and never set otherwise.

```suggestion
interface BaseAggregateResults {
    state: StreamingResultsState
    results: GQL.SearchResult[]
    alert?: Alert
    filters: Filter[]
    progress: Progress
}

interface SuccessfulAggregateResults extends BaseAggregateResults {
    state: 'loading' | 'complete'
}

interface ErrorAggregateResults extends BaseAggregateResults {
    state: 'error'
    error: Error
}

export type AggregateStreamingSearchResults = SuccessfulAggregateResults | ErrorAggregateResults
```

Reply:
Thank you! This is much better.